### PR TITLE
Compatibility with Yoast's polyfills and utils

### PR DESCRIPTION
**Pull request by Screenfeed**
Require the right dependencies:
- The project now requires [yoast/wp-test-utils](https://github.com/Yoast/wp-test-utils).
- `brain/monkey` has been removed, since it is already required by `yoast/wp-test-utils`.
- Added scripts `install-suite` and `install-suite-with-db`: the test suite is required for phpstan to work.

New bash function `downloadWoocommerce()`: this allows to download the plugin from git, with its test suite.

Define the constant `WP_TESTS_PHPUNIT_POLYFILLS_PATH`.

**Improved unit tests (`AbstractTestCase`):**

- Use `Yoast\WPTestUtils\BrainMonkey\YoastTestCase` as base class instead of `PHPUnit\Framework\TestCase`.
- Renamed the `setUp()` and `tearDown()` methods in snake_case.
- Replaced the static property `$mockCommonWpFunctionsInSetUp` by `$stubTranslationFunctions` and `$stubEscapeFunctions`. Remove the method `mockCommonWpFunctions()` (Yoast provides its own methods).

**Bootstrap for unit tests:**
- New `WP_Syntex\Polylang_Phpunit\Unit\bootstrapSuite()` function for consistency with integration tests.

**Improved integration tests (`TestCaseTrait` ):**

- Rename the `setUp()` and `tearDown()` methods in snake_case.

**Bootstrap for integration tests:**

- Changed the order of the parameters of `WP_Syntex\Polylang_Phpunit\Integration\bootstrapSuite()`:
  - The plugins list is now the 3rd parameter: this is to have the same parameters order than the same function from the unit tests.
  - It is now an entry of a more generic `$args` parameter: better if we need more parameters later.

### CODE REVIEW

**Review by Chouby — CHANGES_REQUESTED** [review]

File: UnitTests/Integration/TestCaseTrait.php [hunk 1]
```diff
@@ -50,7 +48,7 @@ public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) { // p
 	 *
 	 * @return void
 	 */
-	public static function wpTearDownAfterClass() {
+	public static function tear_down_after_class() {
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply UnitTests/Integration/TestCaseTrait.php:53]: Damn, I forgot to revert this one.

File: UnitTests/Unit/bootstrap.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Bootstraps the PLL AI unit tests
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply UnitTests/Unit/bootstrap.php:3]: Another copy-paste drama 😅

File: UnitTests/Integration/TestCaseTrait.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Screenfeed) [reply UnitTests/Integration/TestCaseTrait.php]: Done in 8728c8f.

File: UnitTests/Unit/bootstrap.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (Screenfeed) [reply UnitTests/Unit/bootstrap.php]: Done in ba1523c.

File: bin/wp-download-tools.sh [hunk 3]
```diff
@@ -222,6 +222,39 @@ downloadPolylangPro() {
 	cd "$WORKING_DIR"
 }
 
+# Downloads WooCommerce.
+# Writes to stdout whether the plugin has been installed or not.
+#
+# Globals: WC_VERSION, WORKING_DIR, WP_PLUGINS_DIR
+downloadWoocommerce() {
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply bin/wp-download-tools.sh:229]: Done in 23d5b25.

File: composer.json [hunk 4]
```diff
@@ -10,16 +10,16 @@
 	},
 	"require": {
 		"php": ">=5.6",
-		"brain/monkey": "^2.0",
 		"cheprasov/php-cli-args": "^3.0",
-		"phpunit/phpunit": "^5.7 || ^6.5 || ^7.5 || ^8.5 || ^9.5",
-		"symfony/console": "^3.4 || ^4.4"
+		"phpunit/phpunit": "^5.7.21 || ^6.5 || ^7.5",
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply composer.json]: Done in f5b88be.

File: bin/wp-download-tools.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (Screenfeed) [reply bin/wp-download-tools.sh:229]: Improved in 1ecccb6.

File: README.md [hunk 5]
```diff
@@ -226,26 +238,51 @@ Example for your `bootstrap.php` file:
...
 
-Like for integration tests, some helpers are available in your tests, from the `TestCaseTrait` trait.
+namespace WP_Syntex\Polylang_Foobar\Tests\Unit;
+
+use WP_Syntex\Polylang_Phpunit\Unit\AbstractTestCase;
+
+/**
+ * Tests for `WP_Syntex\Polylang_Foobar\barbaz()`.
+ */
+class Barbaz extends AbstractTestCase {$$
```
Comment (Chouby) [inline README.md:272]: `$$` Is it a typo?

**Review by Chouby — CHANGES_REQUESTED** [review]

**Pull request merged**